### PR TITLE
fix Backend Settings

### PR DESCRIPTION
**Pull request by jankrzempek**
https://trello.com/c/kAzjpy2N/117-settings-backend-url

### CODE REVIEW

File: AirCasting/Graph/GraphView.swift [hunk 1]
```diff
@@ -13,6 +13,7 @@ struct GraphView<StatsViewModelType>: View where StatsViewModelType: StatisticsC
     let thresholds: [SensorThreshold]
     @Binding var selectedStream: MeasurementStreamEntity?
     @StateObject var statsContainerViewModel: StatsViewModelType
+    @EnvironmentObject private var urlProvider: UserDefaultsBaseURLProvider
```
Comment (pawelgil) [inline AirCasting/Graph/GraphView.swift:16]: please use protocol (`BaseURLProvider`) as a dependency, not concrete classes

File: AirCasting/Map/AirMapView.swift [hunk 2]
```diff
@@ -20,6 +20,7 @@ struct AirMapView: View {
     @Binding var showLoadingIndicator: Bool
     @State var isUserInteracting = true
     @Binding var selectedStream: MeasurementStreamEntity?
+    @EnvironmentObject private var urlProvider: UserDefaultsBaseURLProvider
```
Comment (pawelgil) [inline AirCasting/Map/AirMapView.swift:23]: please use protocol (`BaseURLProvider`) as a dependency, not concrete classes

File: AirCasting/RootAppView.swift [hunk 3]
```diff
@@ -24,15 +24,13 @@ struct RootAppView: View {
     @EnvironmentObject var microphoneManager: MicrophoneManager
     @EnvironmentObject var lifeTimeEventsProvider: LifeTimeEventsProvider
     @EnvironmentObject var averagingService: AveragingService
+    @EnvironmentObject private var urlProvider: UserDefaultsBaseURLProvider
```
Comment (pawelgil) [inline AirCasting/RootAppView.swift:27]: please use protocol (`BaseURLProvider`) as a dependency, not concrete classes

File: AirCasting/SessionViews/SessionCardView.swift [hunk 4]
```diff
@@ -17,6 +17,7 @@ struct SessionCardView: View {
     @State private var showLoadingIndicator = false
     @ObservedObject var session: SessionEntity
     @EnvironmentObject var selectedSection: SelectSection
+    @EnvironmentObject private var urlProvider: UserDefaultsBaseURLProvider
```
Comment (pawelgil) [inline AirCasting/SessionViews/SessionCardView.swift:20]: please use protocol (`BaseURLProvider`) as a dependency, not concrete classes

File: AirCasting/SessionViews/SessionHeaderView.swift [hunk 5]
```diff
@@ -16,6 +16,7 @@ struct SessionHeaderView: View {
     @State var chevronIndicator = "chevron.down"
     @EnvironmentObject var networkChecker: NetworkChecker
     @EnvironmentObject var bluetoothManager: BluetoothManager
+    @EnvironmentObject private var urlProvider: UserDefaultsBaseURLProvider
```
Comment (pawelgil) [inline AirCasting/SessionViews/SessionHeaderView.swift:19]: please use protocol (`BaseURLProvider`) as a dependency, not concrete classes

File: AirCasting/SessionsSynchronization/Adapters/Services/SessionDownloadService.swift [hunk 6]
```diff
@@ -43,7 +45,8 @@ final class SessionDownloadService: SessionDownstream, MeasurementsDownloadable
     }
     
     private func download(session: SessionUUID, completion: @escaping (Result<SessionsSynchronization.SessionDownstreamData, Error>) -> Void) -> Cancellable {
-        var urlComponents = URLComponents(string: "http://aircasting.org/api/user/sessions/empty.json")!
+        let urlF = urlProvider.baseAppURL.appendingPathComponent("api/user/sessions/update_session.json")
```
Comment (pawelgil) [inline AirCasting/SessionsSynchronization/Adapters/Services/SessionDownloadService.swift:48]: the F in the name is intentional?

File: AirCasting/SessionsSynchronization/Adapters/Services/SessionDownloadService.swift [hunk 7]
```diff
@@ -75,7 +78,8 @@ final class SessionDownloadService: SessionDownstream, MeasurementsDownloadable
     }
     
     func downloadSessionWithMeasurement(uuid: SessionUUID, completion: @escaping (Result<SessionsSynchronization.SessionDownstreamData, Error>) -> Void) -> Cancellable {
-        var urlComponents = URLComponents(string: "http://aircasting.org/api/user/sessions/empty.json")!
+        let urlF = urlProvider.baseAppURL.appendingPathComponent("api/user/sessions/empty.json")
```
Comment (pawelgil) [inline AirCasting/SessionsSynchronization/Adapters/Services/SessionDownloadService.swift:81]: the F in the name is intentional?

**Review by pawelgil — CHANGES_REQUESTED** [review]

File: AirCasting/SessionsSynchronization/Adapters/Services/SessionDownloadService.swift (hunk 6 above)
Comment (jankrzempek) [reply AirCasting/SessionsSynchronization/Adapters/Services/SessionDownloadService.swift:48]: Not really, forgot to change at the end

File: AirCasting/Dashboard/DashboardView.swift [hunk 8]
```diff
@@ -17,7 +17,7 @@ struct DashboardView: View {
     @EnvironmentObject var selectedSection: SelectSection
     @EnvironmentObject var averaging: AveragingService
     @State var isRefreshing: Bool = false
-
+    var urlProvider: BaseURLProvider
```
Comment (pawelgil) [inline AirCasting/Dashboard/DashboardView.swift:20]: why not private let?

File: AirCasting/Graph/GraphView.swift [hunk 9]
```diff
@@ -24,7 +25,7 @@ struct GraphView<StatsViewModelType>: View where StatsViewModelType: StatisticsC
                                   isExpandButtonNeeded: false,
                                   isSensorTypeNeeded: false,
                                   isCollapsed: Binding.constant(false),
-                                  session: session,
+                                  urlProvider: urlProvider, session: session,
```
Comment (pawelgil) [inline AirCasting/Graph/GraphView.swift:28]: newline missing, 2 params in one line

File: AirCasting/Map/AirMapView.swift [hunk 10]
```diff
@@ -38,7 +39,7 @@ struct AirMapView: View {
                                   isExpandButtonNeeded: false,
                                   isSensorTypeNeeded: false,
                                   isCollapsed: Binding.constant(false),
-                                  session: session,
+                                  urlProvider: urlProvider, session: session,
```
Comment (pawelgil) [inline AirCasting/Map/AirMapView.swift:42]: newline missing, 2 params in one line

File: AirCasting/Map/AirMapView.swift [hunk 11]
```diff
@@ -111,7 +111,7 @@ struct Map_Previews: PreviewProvider {
                    session: .mock,
                    showLoadingIndicator: .constant(true),
                    selectedStream: .constant(nil),
-                   sessionStoppableFactory: SessionStoppableFactoryDummy(),
+                   urlProvider: DummyURLProvider(), sessionStoppableFactory: SessionStoppableFactoryDummy(),
```
Comment (pawelgil) [inline AirCasting/Map/AirMapView.swift:114]: newline missing, 2 params in one line

File: AirCasting/RootAppView.swift [hunk 12]
```diff
@@ -28,10 +28,8 @@ struct RootAppView: View {
     let locationTracker = LocationTracker(locationManager: CLLocationManager())
     var sessionSynchronizer: SessionSynchronizer
     let persistenceController: PersistenceController
-    let urlProvider = UserDefaultsBaseURLProvider()
     let networkChecker = NetworkChecker(connectionAvailable: false)
-    
-    
+    var urlProvider: BaseURLProvider
```
Comment (pawelgil) [inline AirCasting/RootAppView.swift:32]: why var instead of let?

File: AirCasting/SessionViews/SessionCardView.swift [hunk 13]
```diff
@@ -17,6 +17,7 @@ struct SessionCardView: View {
     @State private var showLoadingIndicator = false
     @ObservedObject var session: SessionEntity
     @EnvironmentObject var selectedSection: SelectSection
+    var urlProvider: BaseURLProvider
```
Comment (pawelgil) [inline AirCasting/SessionViews/SessionCardView.swift:20]: why var instead of private let?

File: AirCasting/SessionViews/SessionCardView.swift [hunk 14]
```diff
@@ -134,16 +137,17 @@ private extension SessionCardView {
             },
             isExpandButtonNeeded: true,
             isCollapsed: $isCollapsed,
-            session: session,
-            sessionStopperFactory: sessionStoppableFactory, measurementStreamStorage: measurementStreamStorage, sessionSynchronizer: sessionSynchronizer
+            urlProvider: urlProvider, session: session,
```
Comment (pawelgil) [inline AirCasting/SessionViews/SessionCardView.swift:140]: newline missing, 2 params in one line

File: AirCasting/SessionViews/SessionCardView.swift [hunk 15]
```diff
@@ -134,16 +137,17 @@ private extension SessionCardView {
...
+            sessionSynchronizer: sessionSynchronizer
         )
     }
 
     private var measurements: some View {
         _ABMeasurementsView(measurementsViewModel: DefaultSyncingMeasurementsViewModel(measurementStreamStorage: measurementStreamStorage,
                                                                                        sessionDownloader: SessionDownloadService(client: URLSession.shared,
-                                                                                       authorization: UserAuthenticationSession(),
-                                                                                       responseValidator: DefaultHTTPResponseValidator()),
+                                                                                                                                 authorization: UserAuthenticationSession(), responseValidator: DefaultHTTPResponseValidator(), urlProvider: urlProvider),
```
Comment (pawelgil) [inline AirCasting/SessionViews/SessionCardView.swift:150]: is indentation ok here?

File: AirCasting/SessionViews/SessionHeaderView.swift [hunk 16]
```diff
@@ -16,6 +16,7 @@ struct SessionHeaderView: View {
     @State var chevronIndicator = "chevron.down"
     @EnvironmentObject var networkChecker: NetworkChecker
     @EnvironmentObject var bluetoothManager: BluetoothManager
+    var urlProvider: BaseURLProvider
```
Comment (pawelgil) [inline AirCasting/SessionViews/SessionHeaderView.swift:19]: why not let?

File: AirCasting/SessionViews/SessionHeaderView.swift [hunk 17]
```diff
@@ -221,8 +222,10 @@ struct SessionHeader_Previews: PreviewProvider {
     static var previews: some View {
         SessionHeaderView(action: {},
                           isExpandButtonNeeded: true, isCollapsed: .constant(true),
-                          session: SessionEntity.mock,
-                          sessionStopperFactory: SessionStoppableFactoryDummy(), measurementStreamStorage: PreviewMeasurementStreamStorage(), sessionSynchronizer: DummySessionSynchronizer())
+                          urlProvider: DummyURLProvider(), session: SessionEntity.mock,
```
Comment (pawelgil) [inline AirCasting/SessionViews/SessionHeaderView.swift:225]: newline missing, 2 params in one line

File: AirCasting/SessionViews/StandaloneSessionCardView.swift [hunk 18]
```diff
@@ -10,7 +10,7 @@ struct StandaloneSessionCardView: View {
     let sessionSynchronizer: SessionSynchronizer
     let measurementStreamStorage: MeasurementStreamStorage
     @EnvironmentObject private var sdSyncController: SDSyncController
-    @EnvironmentObject private var urlProvider: UserDefaultsBaseURLProvider
+    var urlProvider: BaseURLProvider
```
Comment (pawelgil) [inline AirCasting/SessionViews/StandaloneSessionCardView.swift:13]: why not let?

File: AirCasting/SessionViews/StandaloneSessionCardView.swift [hunk 19]
```diff
@@ -35,7 +35,7 @@ struct StandaloneSessionCardView: View {
             isExpandButtonNeeded: false,
             isMenuNeeded: false,
             isCollapsed: .constant(false),
-            session: session,
+            urlProvider: urlProvider, session: session,
```
Comment (pawelgil) [inline AirCasting/SessionViews/StandaloneSessionCardView.swift:38]: newline missing, 2 params in one line

File: AirCasting/ForgotPassword/InterfaceAdapters/EmailResetPasswordService.swift [hunk 20]
```diff
@@ -7,16 +7,18 @@ final class EmailResetPasswordService: ResetPasswordService {
     
     private let apiClient: APIClient
     private let validator: HTTPResponseValidator
+    private var urlProvider: BaseURLProvider
```
Comment (pawelgil) [inline AirCasting/ForgotPassword/InterfaceAdapters/EmailResetPasswordService.swift:10]: why not let?

File: AirCasting/Graph/GraphView.swift [hunk 21]
```diff
@@ -13,6 +13,7 @@ struct GraphView<StatsViewModelType>: View where StatsViewModelType: StatisticsC
     let thresholds: [SensorThreshold]
     @Binding var selectedStream: MeasurementStreamEntity?
     @StateObject var statsContainerViewModel: StatsViewModelType
+    var urlProvider: BaseURLProvider
```
Comment (pawelgil) [inline AirCasting/Graph/GraphView.swift:16]: why not let?

File: AirCasting/RootAppView.swift [hunk 22]
```diff
@@ -107,9 +105,9 @@ struct MainAppView: View {
     let measurementStreamStorage: MeasurementStreamStorage
     let locationHandler: LocationHandler
     let sdSyncController: SDSyncController
+    var urlProvider: BaseURLProvider
```
Comment (pawelgil) [inline AirCasting/RootAppView.swift:108]: why not let?

File: AirCasting/Map/AirMapView.swift [hunk 23]
```diff
@@ -20,6 +20,7 @@ struct AirMapView: View {
     @Binding var showLoadingIndicator: Bool
     @State var isUserInteracting = true
     @Binding var selectedStream: MeasurementStreamEntity?
+    var urlProvider: BaseURLProvider
```
Comment (pawelgil) [inline AirCasting/Map/AirMapView.swift:23]: why not let?

**Review by pawelgil — CHANGES_REQUESTED** [review]
some unneeded vars + small stylistic stuff

File: AirCasting/Graph/GraphView.swift (hunk 9 above)
Comment (jankrzempek) [reply AirCasting/Graph/GraphView.swift:28]: great eye, have no seen

File: AirCasting/Dashboard/DashboardView.swift (hunk 8 above)
Comment (jankrzempek) [reply AirCasting/Dashboard/DashboardView.swift:20]: no reason why I have not used let and private in many places..

File: AirCasting/SessionViews/SessionCardView.swift (hunk 15 above)
Comment (jankrzempek) [reply AirCasting/SessionViews/SessionCardView.swift:150]: now I think it should be, sorry

**Review by pawelgil — APPROVED** [review]

**Review by smialko — APPROVED** [review]
Looks good to me :)

**Pull request merged**